WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewModel.kt

```diff
@@ -0,0 +1,252 @@
+                showFullscreenError = false,
+                showFullscreenProgress = false,
+                showContent = true,
+                showSkipButton = StringUtils.isEmpty(query),
+                headerUiState = createHeaderUiState(shouldShowHeader(query), headerInfo),
+                searchInputState = createSearchInputUiState(
+                        query,
+                        showProgress = state is Loading,
+                        hint = headerInfo.inputHint
+                ),
+                items = createSuggestionsUiStates(
+                        onRetry = { updateQuery(query) },
+                        data = state.data,
+                        errorFetchingSuggestions = state is Error
+                )
+        )
+    }
+
+    private fun createSuggestionsUiStates(
+        onRetry: () -> Unit,
+        data: List<VerticalModel>,
+        errorFetchingSuggestions: Boolean
+    ): List<VerticalsListItemUiState> {
+        val items: ArrayList<VerticalsListItemUiState> = ArrayList()
+        if (errorFetchingSuggestions) {
+            val errorUiState = VerticalsFetchSuggestionsErrorUiState(
+                    messageResId = R.string.site_creation_fetch_suggestions_failed,
+                    retryButonResId = R.string.button_retry
+            )
+            errorUiState.onItemTapped = onRetry
+            items.add(errorUiState)
+        } else {
+            val lastItemIndex = data.size - 1
+            data.forEachIndexed { index, model ->
+                items.add(VerticalsModelUiState(model.verticalId, model.name, showDivider = index != lastItemIndex))
+            }
+        }
+        return items
+    }
+
+    private fun shouldShowHeader(query: String): Boolean {
+        return StringUtils.isEmpty(query)
+    }
+
+    private fun createHeaderUiState(
+        isVisible: Boolean,
+        headerInfo: DummyVerticalsHeaderInfoModel
+    ): VerticalsHeaderUiState {
+        return VerticalsHeaderUiState(isVisible, headerInfo.title, headerInfo.subtitle)
+    }
+
+    private fun createSearchInputUiState(
+        query: String,
+        showProgress: Boolean,
+        hint: String
+    ): VerticalsSearchInputUiState {
+        return VerticalsSearchInputUiState(
+                hint,
+                showProgress,
+                showClearButton = !StringUtils.isEmpty(query)
+        )
+    }
+
+    data class VerticalsUiState(
+        val showFullscreenError: Boolean,
+        val showFullscreenProgress: Boolean,
+        val showSkipButton: Boolean,
+        val showContent: Boolean,
+        val headerUiState: VerticalsHeaderUiState? = null,
+        val searchInputState: VerticalsSearchInputUiState? = null,
+        val items: List<VerticalsListItemUiState>
+    )
+
+    data class VerticalsSearchInputUiState(
+        val hint: String,
+        val showProgress: Boolean,
+        val showClearButton: Boolean
+    )
+
+    data class VerticalsHeaderUiState(
```

Comment:
This might be an overkill, but I'd have made this class a sealed class with `HIDDEN` object and `VISIBLE` class with `title` and `subtitle`.

Reply:
I'd like to have as little ifs in the fragment as possible -> More on this comment https://github.com/wordpress-mobile/WordPress-Android/pull/8618#discussion_r234928326.